### PR TITLE
fix: assume blocking unless HEAD

### PR DESCRIPTION
**Pull request by ronag**
This is a safer default.

## Status

- [ ] I have read and agreed to the [Developer's Certificate of Origin][cert]
- [ ] Tested
- [ ] Benchmarked (**optional**)
- [ ] Documented
- [ ] Review ready
- [ ] In review
- [ ] Merge ready

[cert]: https://github.com/nodejs/undici/blob/main/CONTRIBUTING.md

### CODE REVIEW

**Review by mcollina — COMMENTED** [review]
Can you adjust the tests?

**Comment by mcollina** [comment]
Does this affect our benchmarks?

**Comment by ronag** [comment]
> Does this affect our benchmarks?

Yes, if they are using pipelining.

**Comment by ronag** [comment]
Fixed benchmarks as well.

File: lib/core/request.js [hunk 1]
```diff
@@ -143,7 +143,7 @@ class Request {
       ? method === 'HEAD' || method === 'GET'
       : idempotent
 
-    this.blocking = blocking == null ? false : blocking
+    this.blocking = blocking ?? this.method !== 'HEAD'
```
Comment (ronag) [inline lib/core/request.js:146]: We could also check range request headers and/or range and/or content-length response header to dynamically determine this.

**Comment by ronag** [comment]
With this change we could probably turn pipelining on by default?

**Comment by ronag** [comment]
Failing tests seem to be flaky and unrelated

**Review by metcoder95 — APPROVED** [review]

**Comment by ronag** [comment]
@mcollina ping

**Pull request merged**